internal/store/driver.go

```diff
@@ -642,8 +852,416 @@ func (d *Driver) calculateHTTPSEdges() map[string]ingressv1alpha1.HTTPSEdge {
+								continue
+							}
+						case gatewayv1.NamespacesFromSelector:
+							if httproute.Namespace != listener.AllowedRoutes.Namespaces.Selector.String() {
+								continue
+							}
+						}
+					}
+
+					// matches our gateway
+					for _, hostname := range httproute.Spec.Hostnames {
+						if string(hostname) != string(*listener.Hostname) {
+							// doesn't match this listener
+							continue
+						}
+						// matches gateway and listener
+						for _, rule := range httproute.Spec.Rules {
+							// TODO: resolve rule.Matches
+							// TODO: resolve rule.Filters
+							// for v0 we will only resolve the first backendRef
+							pathMatch := "/"
+							pathMatchType := "path_prefix"
+							// first match with a path will be accepted as the route's path
+							for _, match := range rule.Matches {
+								if match.Path != nil {
+									pathMatch = *match.Path.Value
+									if *match.Path.Type == gatewayv1.PathMatchExact {
+										pathMatchType = "exact_path"
+									}
+									break
+								}
+							}
+							route := ingressv1alpha1.HTTPSEdgeRouteSpec{
+								Match:     pathMatch,     // change based on the rule.match
+								MatchType: pathMatchType, // change based on rule.Matches
+							}
+
+							// TODO: set with values from rules.Filters + rules.Matches
+							policy, err := d.createEndpointPolicyForGateway(&rule)
+							if err != nil {
+								d.log.Error(err, "error creating policy from HTTPRouteRule", "rule", rule)
+								continue
+							}
+							route.Policy = policy
+
+							for idx, backendref := range rule.BackendRefs {
+								// currently the ingress controller doesn't support weighted backends
+								// so we'll only support one backendref per rule
+								// TODO: remove when tested with multiple backends
+								if idx > 0 {
+									break
+								}
+								// handle backendref
+								refKind := string(*backendref.Kind)
+								if refKind != "Service" {
+									// only support services currently
+									continue
+								}
+
+								refName := string(backendref.Name)
+								//refNamespace := string(*backendref.Namespace)
+								serviceUID, servicePort, err := d.getEdgeBackendRef(backendref.BackendRef, gtw.Namespace)
+								if err != nil {
+									d.log.Error(err, "could not find port for service", "namespace", gtw.Namespace, "service", refName)
+								}
+
+								route.Backend = ingressv1alpha1.TunnelGroupBackend{
+									Labels: d.ngrokLabels(gtw.Namespace, serviceUID, refName, servicePort),
+								}
+
+							}
+							// set different customMetadata for gateways next
+							route.Metadata = d.customMetadata
+
+							edge.Spec.Routes = append(edge.Spec.Routes, route)
+						}
+					}
+				}
+			}
+
+			edgeMap[domainName] = edge
+		}
+	}
+}
+
+func (d *Driver) createEndpointPolicyForGateway(rule *gatewayv1.HTTPRouteRule) (*ingressv1alpha1.EndpointPolicy, error) {
+	var inboundActions *[]ingressv1alpha1.EndpointAction
```

Comment:
how about:
```
type Actions struct {
    endpointAcitons []ingressv1alpha1.EndpointAction
}
```
then carry `*Actions` around with `add` method. Finally, just check if `len > 0` if we wanna enable it?

Reply:
I think the `endpointActions` would also need to be a pointer in that case

Reply:
oooh, I see what you mean. Yeah that would work

Reply:
updated here: https://github.com/ngrok/kubernetes-ingress-controller/pull/343/commits/a941017f1ea73f664f714ae9d6ec268880019b07